lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -162,38 +171,59 @@ export default function getRecipeInterface(
             let doAntiCsrfCheck = options !== undefined ? options.antiCsrfCheck : undefined;
-            const transferMethod = config.getTokenTransferMethod({ req, userContext });
+            const preferredTransferMethod = config.getTokenTransferMethod({ req, userContext });
+
+            let transferMethod = preferredTransferMethod === "MISSING_AUTH_HEADER" ? "header" : preferredTransferMethod;
 
-            let accessToken = getToken(config, req, "access", userContext, transferMethod);
+            let accessToken = getToken(req, "access", transferMethod);
             if (accessToken === undefined) {
+                const fallbackMethod = transferMethod === "cookie" ? "header" : "cookie";
                 // We are checking if we could've gone ahead with validation if the transferMethod was different
                 // However, we don't want to do the fallback here, but force a call to refresh
                 // This is done to ensure that the browsers update the transfermethod in a timely manner (basically on the next API call)
                 // instead of waiting for the session to expire
-                accessToken = getToken(
-                    config,
-                    req,
-                    "access",
-                    userContext,
-                    transferMethod === "cookie" ? "header" : "cookie"
-                );
+                accessToken = getToken(req, "access", fallbackMethod);
                 if (accessToken !== undefined) {
-                    logDebugMessage(
-                        "getSession: Returning try refresh token because the access token was not sent as a " +
-                            transferMethod
-                    );
+                    if (preferredTransferMethod === "MISSING_AUTH_HEADER") {
+                        transferMethod = fallbackMethod;
+                        logDebugMessage(
+                            `getSession: falling back to use ${fallbackMethod} as an authentication-method`
+                        );
+                    } else {
+                        logDebugMessage(
+                            `getSession: returning TRY_REFRESH_TOKEN because preferred auth-mode doesn't allow fallback and access token was sent with the wrong method`
+                        );
+                        throw new STError({
+                            message: "access token sent with the wrong method. Please call the refresh API",
+                            type: STError.TRY_REFRESH_TOKEN,
+                        });
+                    }
+                }
+            }
+
+            if (accessToken === undefined) {
+                // This is done here to ensure smooth migration of sessions started before removing the id-refresh-token
+                // This token isn't handled by getToken to limit the scope of this legacy/migration code
+                if (req.getCookieValue("sIRTFrontend") !== undefined) {
```

Comment:
Since this block is related to migration, i will have a look at it later. I prefer if for now, all code is written without migration in mind and then we can add the migration code in isolation after - leads to easier code reviews.